Resfeber/View Controllers/AddTripViewController.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  AddTripViewController.swift
+//  Resfeber
+//
+//  Created by Joshua Rutkowski on 12/5/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import CoreData
+
+class AddTripViewController: UIViewController {
+    
+    // MARK: - Properties
+    var tripService: TripService!
+    let context = CoreDataStack.shared.mainContext
+
+    fileprivate let tripImage: UIButton = {
+        let diameter: CGFloat = 150
+        let button = UIButton()
+        button.isUserInteractionEnabled = false
+        button.setDimensions(height: diameter, width: diameter * 1.5)
+        button.layer.cornerRadius = 12
+        button.layer.masksToBounds = true
+        button.contentMode = .scaleAspectFill
+        button.backgroundColor = .systemGray3
+
+        let config = UIImage.SymbolConfiguration(pointSize: diameter * 0.8)
+        let placeholderImage = UIImage(systemName: "photo.fill")?
+            .withConfiguration(config)
+            .withTintColor(.systemGray6, renderingMode: .alwaysOriginal)
+        
+        button.setImage(placeholderImage, for: .normal)
+        return button
+    }()
+
+    fileprivate var nameTextField = UITextField()
+    fileprivate var startDateTextField = UITextField()
+    fileprivate var endDateTextField = UITextField()
+    
+    fileprivate lazy var tripInfoStackView: UIStackView = {
+        
+        let sectionTitles = ["Trip Name", "Start Date", "End Date"]
+        let textFields = [nameTextField, startDateTextField, endDateTextField]
+        
+        var verticalStackSubViews = [UIView]()
+        verticalStackSubViews.append(seperatorView())
+        
+        for i in sectionTitles.indices {
+            let label = UILabel()
+            label.font = UIFont.boldSystemFont(ofSize: 14)
+            label.text = sectionTitles[i]
+            label.setDimensions(width: 86)
+            
+            textFields[i].font = UIFont.systemFont(ofSize: 14)
+            textFields[i].textColor = RFColor.red
+            startDateTextField.setInputViewDatePicker(target: self, selector: #selector(tapStartDateDone))
+            endDateTextField.setInputViewDatePicker(target: self, selector: #selector(tapEndDateDone))
+            
+            let hStack = UIStackView(arrangedSubviews: [spacer(width: 20), label, textFields[i], spacer(width: 20)])
+            hStack.axis = .horizontal
+            hStack.alignment = .firstBaseline
+            hStack.spacing = 4
+            verticalStackSubViews.append(hStack)
+            verticalStackSubViews.append(seperatorView())
+        }
+        
+        let stack = UIStackView(arrangedSubviews: verticalStackSubViews)
+        stack.axis = .vertical
+        stack.spacing = 10
+        return stack
+    }()
+
+    // MARK: - Lifecycle
+    
```

Comment:
Set up tripService to be a dependency with a custom initializer:
(Refer to inline comment on line 16)
```suggestion
    
    init(tripService: TripService) {
        self.tripService = tripService
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
```

Reply:
Implemented suggested change in PR #33